src/mozilla_sec_eia/models/sec_eia_record_linkage/transform_sec_input.py

```diff
@@ -0,0 +1,383 @@
+    matched to a 10K filer company.
+    """
+    ex21_df.loc[:, "sec_company_id"] = (
+        ex21_df["parent_company_cik"]
+        + "_"
+        + ex21_df["company_name"]
+        + "_"
+        + ex21_df["location_of_inc"]
+    )
+    return ex21_df
+
+
+@asset(
+    ins={
+        "ex21_dfs": AssetIn("ex21_company_ownership_info"),
+        "sec10k_filing_metadata_dfs": AssetIn("sec10k_filing_metadata"),
+    },
+)
+def transformed_ex21_subsidiary_table(
+    ex21_dfs: dict[str, pd.DataFrame],
+    sec10k_filing_metadata_dfs: dict[str, pd.DataFrame],
+) -> pd.DataFrame:
+    """Transform Ex. 21 table of subsidiaries before combining with basic 10k table."""
+    ex21_df = pd.concat(ex21_dfs.values())
+    sec10k_filing_metadata = pd.concat(sec10k_filing_metadata_dfs.values())
+
+    ex21_df.loc[:, "filename"] = convert_ex21_id_to_filename(ex21_df)
+    ex21_df = ex21_df.drop(columns=["id"])
+    ex21_df = _add_report_year_to_sec(ex21_df, sec10k_filing_metadata)
+    ex21_df = ex21_df.rename(columns=EX21_COL_MAP)
+    ex21_df = clean_location_of_inc(ex21_df)
+    # TODO: what to do with the clean company name?
+    ex21_df = transform_company_name(ex21_df)
+    ex21_df = _add_parent_company_cik(ex21_df, sec10k_filing_metadata)
+    # add an sec_company_id, ultimately this ID become the subsidiary's CIK
+    # if the subsidiary is matched to an SEC filer
+    ex21_df = create_sec_company_id_for_ex21_subs(ex21_df=ex21_df)
+    ex21_df = flatten_companies_across_time(
+        df=ex21_df, key_cols=["sec_company_id"], date_col="report_year"
+    )
+    ex21_df = ex21_df.fillna(np.nan)
+
+    return ex21_df
+
+
+def transform_basic10k_table(
+    basic_10k_df: pd.DataFrame, sec10k_filing_metadata: pd.DataFrame
+) -> pd.DataFrame:
+    """Transformations on SEC basic 10K filer table to prepare for record linkage."""
+    basic_10k_df = basic_10k_df.reset_index().pivot_table(
+        values="value", index="filename", columns="key", aggfunc="first"
+    )
+    basic_10k_df.columns.name = None
+    basic_10k_df = (
+        basic_10k_df.reset_index()
+        .pipe(_remove_weird_sec_cols)
+        .pipe(_add_report_year_to_sec, sec10k_filing_metadata)
+        .rename(columns=SEC_COL_MAP)
+        .pipe(clean_location_of_inc)
+        .pipe(transform_company_name)
+        .assign(
+            zip_code=lambda df: df["zip_code"].str[:5],
+            files_10k=True,
+            sec_company_id=lambda df: df["central_index_key"],
+        )
+        .pipe(fill_street_address_nulls)
+    )
+    basic_10k_df[STR_COLS] = basic_10k_df[STR_COLS].apply(
+        lambda x: x.str.strip().str.lower()
+    )
+    basic_10k_df["street_address"] = expand_street_name_abbreviations(
+        basic_10k_df["street_address"]
+    )
+    # flatten across time on unique company name and address pair
+    basic_10k_df = flatten_companies_across_time(
+        df=basic_10k_df, key_cols=["company_name", "street_address"]
+    )
```

Comment:
Instead of flattening across time to have one record per CIK, I opted to flatten so that there is one record per unique company name and street address pair. This is because it's helpful to have all address changes in the record linkage process, in case the company is still reporting an old address to EIA. Later, in the output table, I think we should flatten across time, so there's one record per `sec_company_id`, but maybe it's not a big deal to have that key be unique